rfc/rfc004-distributed-document-format.md

```diff
@@ -0,0 +1,120 @@
+# RFC004 Distributed Document Format 
+
+|  |  |
+| :--- | :--- |
+| Nuts foundation | R.G. Krul |
+| Request for Comments: 004 | Nedap |
+|  | September 2020 |
+
+## Distributed Document Format
+
+### Abstract
+
+### Status of document
+
+This document is currently a draft.
+
+### Copyright Notice
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1.  Introduction
+
+This RFC describes an interoperable, content agnostic data format for distributed networks which provides
+cryptographic authentication and verification of its contents, guaranteed ordering and access control.      
+
+## 2. Terminology
+
+* **Document**: piece of application data enveloped with metadata like signatures and references to other documents
+  published on a distributed network.
+
+Other terminology comes from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Requirements
+The following requirements apply for the document format:
+
+* **Flexible content**: the format must support any kind of payload, textual or binary, large or small.
+* **Consistent ordering**: since documents might revise or require a previous document, there must be ordering that's consistent over the network.
+* **Integrity protected**: since second- or third parties MAY relay documents, document integrity MUST be cryptographically verifiable.
+* **Authenticity protected**: for the same reason as above, document authenticity (who authored the document) MUST be cryptographically verifiable.
+* **Anonymous Confidentiality Control**: since some documents MAY be confidential (just to be read by specific parties)
+  there MUST be a way to indicate that the document is confidential. Which party (or parties) is allowed to read the document
+  sensitive information in itself, and should only be disclosed to the involved parties.
+  In other words if party A publishes a document only to be read by party B, only they should see that fact. Uninvolved
+  party C MAY only see a confidential document from party A but not the involvement of party B. 
+  (TBD: should this actually be here yet?)
+
+## 4. Document Format
+The [RFC7159 JavaScript Object Notation](https://tools.ietf.org/html/rfc7159) (JSON) MUST be used as base format.
+
+### 4.1. Document Structure
+
+A document MUST contain the following fields:
+* **ref**: (reference) MUST contain the reference (see section 4.2) of the document as hexadecimal string.
```

Comment:
Why put it in there? Just define the ref as the hash over the normalized json. That way it can't be wrongly entered either.

Reply:
I went back and forth a few times, but I'll remove it and make the actual document the payload of the JWS itself. Then we'll have to do a migration of existing data though.